src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -100,30 +151,19 @@ public boolean equals(Object other) {
         Lesson otherLesson = (Lesson) other;
         return date.equals(otherLesson.date)
                 && time.equals(otherLesson.time)
-                && students.equals(otherLesson.students);
+                && students.equals(otherLesson.students)
+                && attendanceMap.equals(otherLesson.attendanceMap);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(date, time, students);
+        return Objects.hash(date, time, students, attendanceMap);
     }
 
     @Override
     public String toString() {
-        return String.format("Lesson[date=%s, time=%s, students=%s]", date, time, students);
+        return String.format("Lesson[date=%s, time=%s, students=%s, attendanceMap=%s]",
```

Comment:
Maybe you could add a simple test for the toString() method, just to clear the CodeCov?